aiounifi/devices.py

```diff
@@ -345,3 +384,77 @@ def __getitem__(self, obj_id: int) -> Port:
     def __iter__(self) -> Iterator[Union[int, str]]:
         """Iterate over ports."""
         return iter(self.ports)
+
+
+class Outlet:
```

Comment:
Can most properties of outlet be expected to exist?, then you can replace all raw.get with raw[]

Reply:
I changed name, index, has_relay, and has_metering to use self.raw[] instead of self.raw.get().  I think relay_state might be optional depending on if the outlet has a relay or not.  I also added another optional property "cycle_enabled".  cycle_enabled seems to show up by default on the UP6, but is only present on the UP1 after flipping the "Modem Power Cycle" switch in the UniFi Network controller.

Reply:
Ok, what does cycle_enabled do?

Reply:
It's used to indicate which outlet(s) your modem or other internet equipment is connected to.  If you have a UniFi Security Gateway, or a Dream Machine, or some other UniFi router, the router will inform the network controller when the internet goes down.  If you have one or more outlets with cycle_enabled=True, the UniFi network controller will power cycle those outlets to try and regain internet connectivity.  For the purposes of this library, it's not super important and is really just informational.  Maybe someone wants to use that flag to have hass restart their modem under certain conditions?

Reply:
As a separate PR, could you look into how to set cycle_enabled True/False resulting in a method to do that?

Reply:
I'd be happy to.